tutorials/klusterai-api/effective-prompt-engineering.ipynb

```diff
@@ -0,0 +1,589 @@
+    "\n",
+    "Please be concise, specific, and actionable.\"\"\"\n",
+    "\n",
+    "# Display the prompt for visual reference\n",
+    "display(Markdown(best_prompt_text))\n",
+    "\n",
+    "# Use the variable in the API call\n",
+    "response_best = client.chat.completions.create(\n",
+    "    model=model,\n",
+    "    messages=[{\"role\": \"user\", \"content\": best_prompt_text}],\n",
+    "    temperature=0.5\n",
+    ").choices[0].message.content\n",
+    "print(response_best)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3455c366",
+   "metadata": {},
+   "source": [
+    "### Analysis\n",
+    "The Best prompt excels because it integrates comprehensive data (product sales, prices, and profit margins) with detailed business context (target demographics, product characteristics, marketing channels, and specific challenges). By explicitly specifying the desired structure—including step-by-step revenue and profit analysis, targeted profitability recommendations, and marketing optimizations—it guides the LLM toward high-value, actionable insights.\n",
+    "\n",
+    "Key strengths of this Best prompt:\n",
+    "1. **Complete Data & Metrics**\n",
+    "   * Monthly sales, pricing, and profit margins remove guesswork and allow precise revenue/profit calculations.\n",
+    "2. **Focused Business Context**\n",
+    "   * Target customer demographics, marketing channels, and product nuances (e.g., overstock of Product C) align the output with real business priorities.\n",
+    "3. **Clear Task Breakdown**\n",
+    "   * Dividing the request into distinct sections (Revenue/Profit Analysis, Profitability Recommendations, Marketing Strategies) ensures the response is both methodical and thorough.\n",
+    "4. **Actionability & Specificity**\n",
+    "   * The prompt requests concrete next steps—such as pricing tweaks and promotional ideas—tailored to each product's characteristics and customer segment.\n",
+    "5. **Logical Reasoning Instructions**\n",
+    "   * Guiding the LLM to \"think step-by-step\" fosters more thoughtful planning, especially around clearing excess inventory and leveraging different marketing channels.\n",
+    "\n",
+    "Overall, this approach maximizes the LLM's potential, yielding responses that are precise, context-aware, and immediately useful to decision-makers."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7a0583b5",
+   "metadata": {},
+   "source": [
+    "## Conclusion\n",
+    "\n",
+    "This side-by-side comparison demonstrates how the quality and specificity of a prompt directly influence the depth, utility, and strategic value of an LLM's output:\n",
+    "* **Bad Prompt:** Minimal data and vague instructions yield only superficial insights—no clear direction or actionable strategies.\n",
+    "* **Better Prompt:** Adding pricing details and specific tasks (like calculating revenue or ranking products) results in more concrete analysis and targeted recommendations.\n",
```

Comment:
Remember that in kluster we dont use Camel Case capitalization, please change here throughout.